workers/unity/Packages/io.improbable.gdk.core/Subscriptions/SubscriptionAggregate.cs

```diff
@@ -44,24 +42,21 @@ public ISubscriptionAvailabilityHandler GetAvailabilityHandler()
 
         public T GetValue<T>()
         {
-            if (subscriptionsSatisfied != subscriptions.Length)
-            {
-                throw new InvalidOperationException("Subscriptions not all satisfied");
-            }
-
-            if (!typesToSubscriptionIndexes.TryGetValue(typeof(T), out var subscriptionIndex))
-            {
-                throw new InvalidOperationException("No subscription with that type");
-            }
+            var subscriptionIndex = GetTypeIndex(typeof(T));
+            var subscription = (Subscription<T>) subscriptions[subscriptionIndex];
 
-            var sub = (Subscription<T>) subscriptions[subscriptionIndex];
-
-            return sub.Value;
+            return subscription.Value;
         }
 
         public object GetErasedValue(Type type)
         {
-            if (subscriptionsSatisfied != subscriptions.Length)
+            var subscriptionIndex = GetTypeIndex(type);
+            return subscriptions[subscriptionIndex].GetErasedValue();
+        }
+
+        private int GetTypeIndex(Type type)
```

Comment:
Feels weird that this throws if things aren't satisified. 

Looking purely at the name & signature of this function, that is entirely surprising